Lib/logging/__init__.py

```diff
@@ -1620,7 +1620,7 @@ def findCaller(self, stack_info=False, stacklevel=1):
                 sinfo = sio.getvalue()
                 if sinfo[-1] == '\n':
                     sinfo = sinfo[:-1]
-        return co.co_filename, f.f_lineno, co.co_name, sinfo
+        return co.co_filename, f.f_lineno, co.co_name, sinfo, co.co_qualname
```

Comment:
findCaller is documented, public API.
This breaks backward compatibility.

Reply:
Oh. Missed that.
So it cannot change the return type. 

What could be the best alternative ?
Is introducing another method such as find_caller2 an option ?

Reply:
No. 3rd party code may override findCaller. Not calling it also breaks backward compatibility.

I think there is no way to add qualname in Logger without breaking backward compatibility.
You should create custom logger if you want qualname.

Reply:
This is an example that customize findCaller.

https://github.com/hynek/structlog/blob/b488a8bf589a01aabc41e3bf8df81a9848cd426c/src/structlog/stdlib.py#L107

I think returning `co.co_qualname` instead of `co.co_name` is the best way to customize Logger.

Maybe, the Logger can have `use_qualname` attribute.

```python
logger = logging.getLogger(__name__)
logger.use_qualname = True  # logger.findCaller() returns co_qualname instead of co_name
```